File: SmartPhone++/SmartPhone.Services/Database/PhoneModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SmartPhone.Services.Database
{
    public class PhoneModel
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string Brand { get; set; } = string.Empty; // Apple, Samsung, Xiaomi, etc.
        
        [Required]
        [MaxLength(100)]
        public string Model { get; set; } = string.Empty; // iPhone 15 Pro, Galaxy S24 Ultra, etc.
        
        [MaxLength(100)]
        public string? Series { get; set; } // Pro, Ultra, Plus, etc.
        
        [MaxLength(50)]
        public string? Year { get; set; }
        
        [MaxLength(100)]
        public string? Color { get; set; }
        
        [MaxLength(50)]
        public string? Storage { get; set; } // 128GB, 256GB, etc.
        
        [MaxLength(50)]
        public string? RAM { get; set; } // 8GB, 12GB, etc.
        
        [MaxLength(100)]
        public string? Network { get; set; } // 4G, 5G, etc.
        
        public string? ImageUrl { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime? UpdatedAt { get; set; }
        
        // Navigation properties
        public ICollection<PartCompatibility> CompatibleParts { get; set; } = new List<PartCompatibility>();
        public ICollection<Service> Services { get; set; } = new List<Service>();
    }
} 

File: SmartPhone++/UI/smartphone_desktop_admin/lib/model/phone_model.dart
import 'package:json_annotation/json_annotation.dart';

part 'phone_model.g.dart';

@JsonSerializable()
class PhoneModel {
  final int id;
  final String brand;
  final String model;
  final String? series;
  final String? year;
  final String? color;
  final String? storage;
  final String? ram;
  final String? network;
  final String? imageUrl;
  final bool isActive;
  final DateTime createdAt;
  final DateTime? updatedAt;

  PhoneModel({
    required this.id,
    required this.brand,
    required this.model,
    this.series,
    this.year,
    this.color,
    this.storage,
    this.ram,
    this.network,
    this.imageUrl,
    required this.isActive,
    required this.createdAt,
    this.updatedAt,
  });

  factory PhoneModel.fromJson(Map<String, dynamic> json) => _$PhoneModelFromJson(json);
  Map<String, dynamic> toJson() => _$PhoneModelToJson(this);
} 